File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    
    dp = [0]*(number+1)
    
    for i in range(1,number+1):
        for j in range(1,int(i**(1/2))+1):
            if i%j == 0:
                dp[i] +=1
                if j < i//j:
                    dp[i] +=1
        if dp[i] > limit:
            dp[i] = power
        answer += dp[i]
    
    
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        List<Integer> arr = new ArrayList<>();
        
        
        for(int i=1; i<=number;i++){
            int cnt = 0;
            for (int j=1; j*j<=i;j++){
                if(j*j==i) cnt++;
                else if(i%j==0) cnt+=2;
            }
            answer += cnt > limit ? power : cnt;
        }
        return answer;
    }
}